### DISCUSSION TITLE
OpenGL ES render weather grib data

### DISCUSSION THREAD

**Discussion by qq908323236** [mapsforge/vtm#938] [Ideas]
Hello,

I want to render weather grid data through OpenGLES，Like this
![image](https://user-images.githubusercontent.com/19360225/176806539-92f75d79-b4b4-4325-a556-1c2575133df8.png)
Now the weather data is available,

I don't know how to transform opengles vertex coordinates into map coordinates，I didn't find this matrix.

Vertex coordinates:
![image](https://user-images.githubusercontent.com/19360225/176807528-29ba770c-abbe-43bf-9710-435ae657059d.png)

I want to use a matrix to make the rendered position of vertex coordinates cover the whole map.

**Comment by devemux86** [comment]
It is not clear what you mean.

To display maps, vector or bitmap, you can see the samples ([Android](https://github.com/mapsforge/vtm/tree/master/vtm-android-example) or [Desktop](https://github.com/mapsforge/vtm/tree/master/vtm-playground)).

**Reply by qq908323236** [reply]
I've read the samples, but I can't find what I want.
That is, I want to render a layer on the map through OpenGLES, and then this layer should also be scaled with the map at the same time.
What I want is to transfer the four points ((0,0), (0,1), (1,1), (1,0)) of Mercator coordinate system to the shader as vertex coordinates, and then transfer the matrix to the shader so that the vertex coordinates can correspond to the map.
I don't know how to create this matrix.

**Reply by CodeK1988** [reply]
@qq908323236 hi bro, did you solve it？
